gui/frontend/src/search/vis/TextWithAnnotations.js

```diff
@@ -0,0 +1,45 @@
+
+var TextWithAnnotations = (function() {
+  var create = function(obj) {
+    content = obj._source.content
+
+    var el = $('<div></div>')
+        .append($('<span></span>'))
+            .attr({'style':'white-space:pre-wrap'})
+            .append(jQuery.parseHTML(content))
+
+    // compute correct char offsets when there
+    // are HTML tags (which we interpret)
+    var curchar = 0
+    var charmap = []
+    var in_tag = false
+    for (var i=0; i < content.length; i++) {
```

Comment:
This feels very brittle... The best approach would be to feed original content to the parser. The parser supports html; it just would choke on malformed xml tags (e.g., unmatched closing tag) -- not sure if that issue still exists today.